subsys/bluetooth/rpc/client/bt_rpc_conn_client.c

```diff
@@ -1624,3 +1680,11 @@ bt_security_t bt_conn_get_security(struct bt_conn *conn)
 	return BT_SECURITY_L1;
 }
 #endif /* defined(CONFIG_BT_SMP) */
+
+void bt_rpc_conn_init(void)
+{
+	STRUCT_SECTION_FOREACH(bt_conn_cb, cb) {
+		bt_conn_cb_register_on_remote();
```

Comment:
It is for callbacks located in special section. They are not registered, but automatically handled by bluetooth stack. If we have at least one such structure in section on application side, then we have to register callbacks on remote side at initialization time.

Reply:
I understand but isn't not enough to call `bt_conn_cb_register_on_remote` once if there is at least one element in this section ? and next if we register callback on network side here do we need call serialized API for `bt_conn_cb_register` ?

Reply:
But that how it works, it calls bt_conn_cb_register_on_remote only once (see `return` below). We cannot call `bt_conn_cb_register`, because the structure is in flash and `bt_conn_cb_register` modifies it, so we will end with cpu fault.

On the other hand, I see that in current implementation if user is using section and after that he also calls `bt_conn_cb_register` then `bt_conn_cb_register_on_remote` will be called twice. I will investigate if it is a problem and fix it if yes.

Reply:
I updated `bt_conn_cb_register_on_remote`. It can now be called twice.